receiver/collectdreceiver/receiver.go

```diff
@@ -28,60 +26,69 @@ var _ receiver.Metrics = (*collectdReceiver)(nil)
 // collectdReceiver implements the receiver.Metrics for CollectD protocol.
 type collectdReceiver struct {
 	logger             *zap.Logger
-	addr               string
 	server             *http.Server
 	defaultAttrsPrefix string
 	nextConsumer       consumer.Metrics
 	obsrecv            *receiverhelper.ObsReport
+	createSettings     receiver.CreateSettings
+	config             *Config
 }
 
 // newCollectdReceiver creates the CollectD receiver with the given parameters.
 func newCollectdReceiver(
 	logger *zap.Logger,
-	addr string,
-	timeout time.Duration,
+	cfg *Config,
 	defaultAttrsPrefix string,
 	nextConsumer consumer.Metrics,
 	createSettings receiver.CreateSettings) (receiver.Metrics, error) {
 	if nextConsumer == nil {
 		return nil, component.ErrNilNextConsumer
 	}
-	obsrecv, err := receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{
-		ReceiverID:             createSettings.ID,
-		Transport:              "http",
-		ReceiverCreateSettings: createSettings,
-	})
-	if err != nil {
-		return nil, err
-	}
+
 	r := &collectdReceiver{
 		logger:             logger,
-		addr:               addr,
 		nextConsumer:       nextConsumer,
 		defaultAttrsPrefix: defaultAttrsPrefix,
-		obsrecv:            obsrecv,
-	}
-	r.server = &http.Server{
-		Addr:         addr,
-		Handler:      r,
-		ReadTimeout:  timeout,
-		WriteTimeout: timeout,
+		config:             cfg,
+		createSettings:     createSettings,
 	}
-	return r, err
+	return r, nil
 }
 
 // Start starts an HTTP server that can process CollectD JSON requests.
 func (cdr *collectdReceiver) Start(_ context.Context, host component.Host) error {
+	var err error
+	cdr.server, err = cdr.config.HTTPServerSettings.ToServer(host, cdr.createSettings.TelemetrySettings, cdr)
+	if err != nil {
+		return err
+	}
+	cdr.server.ReadTimeout = cdr.config.Timeout
+	cdr.server.WriteTimeout = cdr.config.Timeout
+	cdr.obsrecv, err = receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{
+		ReceiverID:             cdr.createSettings.ID,
+		Transport:              "http",
+		ReceiverCreateSettings: cdr.createSettings,
+	})
```

Comment:
Why is this moved to start?

Reply:
there was something about having a clean receiver instantiation without too much fuss, and we cache createSettings as a field now with the new way to create the server, so it made sense to move all to Start.

Reply:
Ok, it just seems unrelated to the PR